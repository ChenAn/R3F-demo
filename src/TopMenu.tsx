/** @jsxImportSource @emotion/react */
import React from 'react';
import { Link } from 'react-router-dom';
import { css } from '@emotion/react';

const styles = {
  container: css({
    display: 'flex',
    flexDirection: 'column',
  }),
};

const TopMenu = () => {
  
  return (
    <div css={styles.container}>
      <Link to='/scene1'>サンプル(Three.js)</Link>
      <Link to='/scene2'>サンプル(R3F)</Link>
      <Link to='/scene3'>マウスクリック(Three.js)</Link>
      <Link to='/scene4'>マウスクリック(R3F)</Link>
      <Link to='/scene5'>表面に沿ってドラッグ(R3F)</Link>
      <Link to='/scene6'>3DとDOMを分離</Link>
      <Link to='/scene7'>BillboardとText</Link>
    </div>
  );
};

export default TopMenu;
