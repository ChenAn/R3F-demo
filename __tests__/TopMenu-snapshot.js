import React from 'react';
import { render } from '@testing-library/react';
import TopMenu from '@src/TopMenu'
import { MemoryRouter } from 'react-router-dom';

describe('TopMenu', () => {
  it('初期表示', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <TopMenu/>,
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
