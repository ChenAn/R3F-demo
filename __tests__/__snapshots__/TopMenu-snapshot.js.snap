// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopMenu 初期表示 1`] = `
<DocumentFragment>
  <div
    class="css-1tbn94p-TopMenu"
  >
    <a
      href="/scene1"
    >
      サンプル(Three.js)
    </a>
    <a
      href="/scene2"
    >
      サンプル(R3F)
    </a>
    <a
      href="/scene3"
    >
      マウスクリック(Three.js)
    </a>
    <a
      href="/scene4"
    >
      マウスクリック(R3F)
    </a>
    <a
      href="/scene5"
    >
      表面に沿ってドラッグ(R3F)
    </a>
    <a
      href="/scene6"
    >
      3DとDOMを分離
    </a>
  </div>
  ,
</DocumentFragment>
`;
